Site/admin/components/Comment/Index.php

```diff
@@ -1,384 +1,322 @@
-	 */
-	protected function hasState()
-	{
-		return isset($this->app->session->{$this->getKey()});
-	}
-
-	// }}}
-	// {{{ protected function getKey()
-
-	protected function getKey()
-	{
-		return $this->source.'_search_state';
-	}
-
-	// }}}
-
-	// build phase
-	// {{{ protected function buildInternal()
-
-	protected function buildInternal()
-	{
-		parent::buildInternal();
-		$this->buildMessages();
-		$this->buildSearchForm();
-		$this->buildCommentReplicator();
-	}
-
-	// }}}
-	// {{{ protected function buildCommentReplicator()
-
-	protected function buildCommentReplicator()
-	{
-		$comment_replicator = $this->ui->getWidget('comment_replicator');
-		foreach ($comment_replicator->replication_ids as $id) {
-			$comment_display = $comment_replicator->getWidget('comment', $id);
-			$comment_display->setComment($this->comments[$id]);
-		}
-	}
-
-	// }}}
-	// {{{ protected function buildSearchForm()
-
-	protected function buildSearchForm()
-	{
-		$form = $this->ui->getWidget('search_form', true);
-		$form->action = $this->source;
-	}
-
-	// }}}
+    public const SHOW_UNAPPROVED = 1;
+    public const SHOW_ALL = 2;
+    public const SHOW_ALL_SPAM = 3;
+    public const SHOW_SPAM = 4;
+
+    /**
+     * @var SiteCommentDisplay
+     */
+    protected $comment_display;
+
+    /**
+     * @var string
+     */
+    protected $where_clause;
+
+    /**
+     * @var array
+     */
+    protected $comments;
+
+    // init phase
+
+    protected function initInternal()
+    {
+        parent::initInternal();
+
+        $this->ui->loadFromXML($this->getUiXml());
+
+        $this->comment_display = $this->getCommentDisplayWidget();
+
+        $this->ui->getWidget('comment_replicator')->add(
+            $this->comment_display
+        );
+
+        $visibility = $this->ui->getWidget('search_visibility');
+        $visibility->addOptionsByArray($this->getVisibilityOptions());
+        $visibility->value = $this->getDefaultVisibilityValue();
+
+        $this->processSearchUi();
+
+        $this->initComments();
+        $this->initCommentReplicator();
+    }
+
+    protected function getUiXml()
+    {
+        return __DIR__ . '/index.xml';
+    }
+
+    protected function getTable()
+    {
+        return 'Comment';
+    }
+
+    protected function initComments()
+    {
+        $pager = $this->ui->getWidget('pager');
+        $pager->total_records = $this->getCommentCount();
+
+        $comments = $this->getComments(
+            $pager->page_size,
+            $pager->current_record
+        );
+
+        $this->comments = [];
+        foreach ($comments as $comment) {
+            $this->comments[$comment->id] = $comment;
+        }
+
+        // init result message
+        $visibility = $this->ui->getWidget('search_visibility')->value;
+        switch ($visibility) {
+            default:
+            case self::SHOW_UNAPPROVED :
+                $this->ui->getWidget('results_message')->content =
+                    $pager->getResultsMessage(
```

Comment:
can/should this change to a match expression? It looks like the expression is the same in each case with a different string passed in.